### PR TITLE
[PowerPC] Forbid f128 SELECT_CC optimized into fsel

### PR DESCRIPTION
**Pull request by ecnelises**
fsel only supports f32 and f64.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 1]
```diff
@@ -8080,7 +8080,8 @@ SDValue PPCTargetLowering::LowerSELECT_CC(SDValue Op, SelectionDAG &DAG) const {
   // For more information, see section F.3 of the 2.06 ISA specification.
   // With ISA 3.0
   if ((!DAG.getTarget().Options.NoInfsFPMath && !Flags.hasNoInfs()) ||
-      (!DAG.getTarget().Options.NoNaNsFPMath && !Flags.hasNoNaNs()))
+      (!DAG.getTarget().Options.NoNaNsFPMath && !Flags.hasNoNaNs()) ||
+      ResVT == MVT::f128)
```
Comment (lei137) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:8084]: does that mean fsel also does not support `ppc_fp128`?

Comment (ecnelises) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:8084]: yes, we don't have instruction for `ppc_fp128` fsel. The motivating case exists from https://github.com/llvm/llvm-project/commit/097a95f2df4666df74909ab1d3bcc0c1824ace86 , which sets `select_cc f128` as custom, while we don't have that lowering for `ppc_fp128`.

Comment (chenzheng1030) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:8084]: But https://github.com/llvm/llvm-project/commit/097a95f2df4666df74909ab1d3bcc0c1824ace86 will not try to custom `SELECT_CC` to fsel, right? That commit is only for ieee long double.
Do we have other pattern that will generate fsel operating on ppc_fp128 type?

Comment (ecnelises) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:8084]: No. `PPCISD::FSEL` is only generated in `LowerSELECT_CC`, while `SELECT_CC ppc_fp128` is expand.

**Review by chenzheng1030 — APPROVED** [review]
This seems reasonable to me. 
Please wait for one day in case @lei137 has other concerns. Thanks for fixing this.

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp (hunk 1 above)
Comment (lei137) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:8084]: Thank-you!

**Review by lei137 — APPROVED** [review]

**Pull request merged**